### PR TITLE
fix(efs): fixed kmsKeyId to use ARN instead of ID

### PR DESCRIPTION
**Pull request by gadiener**
### Description of your changes

When referencing a Crossplane created KMS key using `kmsKeyIdRef` to a `filesystem.efs.aws.upbound.io` resource the provider is pulling the ID of the key, but the EFS expects the ARN.

Error:
```
Warning  CannotObserveExternalResource    7s (x3 over 72s)     managed/efs.aws.upbound.io/v1beta1, kind=filesystem  cannot run refresh: refresh failed: "kms_key_id" (1bdb68ee-c024-4003-8500-aa5e6a1d86ab) is an invalid ARN: arn: invalid prefix
```

Refer https://github.com/upbound/provider-aws/issues/637 for similar issue.

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Created example resources in EKS:

```
NAME                                                  READY   SYNCED   EXTERNAL-NAME          AGE
filesystem.efs.aws.upbound.io/example-efs-encrypted   True    True     fs-0708fdc7bxxxxxxxx   10m

NAME                                 READY   SYNCED   EXTERNAL-NAME                          AGE
key.kms.aws.upbound.io/example-key   True    True     6a7cc54e-7767-4e01-80c1-xxxxxxxxxxxx   28m
```

```
kubectl get filesystem.efs.aws.upbound.io/example-efs-encrypted -o jsonpath="{.spec.forProvider.kmsKeyId}"
arn:aws:kms:eu-west-1:xxxxxxxxxxxx:key/6a7cc54e-7767-4e01-80c1-xxxxxxxxxxxx  
```

Uptest: https://github.com/upbound/provider-aws/actions/runs/5606583248

### CODE REVIEW

File: examples/efs/filesystem_encrypted.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+apiVersion: efs.aws.upbound.io/v1beta1
+kind: FileSystem
+metadata:
```
Comment (turkenf) [inline examples/efs/filesystem_encrypted.yaml:3]:
```suggestion
metadata:
  annotations:
    meta.upbound.io/example-id: efs/v1beta1/filesystem
```
Could you please add the example id?

File: examples/efs/filesystem_encrypted.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    region: us-west-1
+    creationToken: my-product
+    tags:
+      Name: MyProduct
+    kmsKeyIdRef:
+      name: efs-key
+---
+apiVersion: kms.aws.upbound.io/v1beta1
+kind: Key
+metadata:
```
Comment (turkenf) [inline examples/efs/filesystem_encrypted.yaml:18]:
```suggestion
metadata:
  annotations:
    meta.upbound.io/example-id: efs/v1beta1/filesystem
```
Could you please add the example id?

File: examples/efs/filesystem_encrypted.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+apiVersion: efs.aws.upbound.io/v1beta1
+kind: FileSystem
+metadata:
+  name: example
+  labels:
+    testing.upbound.io/example-name: efs
+spec:
+  forProvider:
+    region: us-west-1
```
Comment (turkenf) [inline examples/efs/filesystem_encrypted.yaml:9]:
```suggestion
    region: us-west-1
    encrypted: true
```
When specifying `kms_key_id`, encrypted needs to be set to true: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/efs_file_system#kms_key_id

**Review by turkenf — COMMENTED** [review]
Thank you for raising this PR @gadiener. I left a few comments on the PR, please consider the following if you want to move this PR forward:
- Please sign the CLA
- Please change the example name from `filesystem_encrypted.yaml` to `filesystem-with-encrypted.yaml`
- Please fill in the description properly(How has this code been tested)

FYI: Next release of official providers is scheduled for Thu, 27 July.

**Comment by gadiener** [comment]
Thanks @turkenf the PR is still in draft, I'm testing it as we speak. Thanks for the early comments!

**Comment by gadiener** [comment]
all right @turkenf, everything should be fixed now. Let me know if you have any additional comment

**Comment by turkenf** [comment]
/test-examples="examples/efs/filesystem-with-encrypted.yaml"

File: config/kms/config.go [hunk 4]
```diff
@@ -33,6 +33,13 @@ func Configure(p *config.Provider) {
 		}
 	})
 
+	p.AddResourceConfigurator("aws_efs_file_system", func(r *config.Resource) {
+		r.References["kms_key_id"] = config.Reference{
+			Type:      "github.com/upbound/provider-aws/apis/kms/v1beta1.Key",
+			Extractor: common.PathARNExtractor,
+		}
+	})
```
Comment (turkenf) [inline config/kms/config.go:41]: Could you please move this config to https://github.com/upbound/provider-aws/blob/main/config/efs/config.go as the main resource is in the efs group?

Comment (gadiener) [reply config/kms/config.go]: Makes sense, done.

Tested again and it looks good

```
NAME                                                  READY   SYNCED   EXTERNAL-NAME          AGE
filesystem.efs.aws.upbound.io/example-efs-encrypted   True    True     fs-040c785a941xxxxxx   12m

NAME                                 READY   SYNCED   EXTERNAL-NAME                          AGE
key.kms.aws.upbound.io/example-key   True    True     d4b0e1dc-7795-4631-be1f-xxxxxxxxxxxx   12m
```

**Review by turkenf — APPROVED** [review]
Thank you so much for your contribution and quick updates @gadiener.

**Pull request merged**